docs/getting-started/istio/index.md

```diff
@@ -33,15 +33,19 @@ spec:
           virtualService:
             # Reference to a VirtualService which the controller updates with canary weights
             name: rollouts-demo-vsvc
+            # optional if there is a single route in VirtualService, required otherwise
             routes:
-            - primary # optional if there is a single route in VirtualService, required otherwise
+            - http-primary
+            - https-3000
```

Comment:
It seems the approach you are taking relies on convention and prefixing to determine if we need to modify HTTP routes vs TLS routes. Instead of convention, can we take a more explicit approach and introduce a new `tlsRoutes` field? Istio VirtualService also separates the two and I feel this would match their spec better.

The existing `routes` would continue to modify only HTTP routes, matching by name, and the new tlsRoutes would match by port numbers. The spec would look like this:

```yaml
          virtualService:
            name: rollouts-demo-vsvc
            routes:
            - primary
            tlsRoutes:
            - 3000
```

Reply:
Sure, if you think that's cleaner then we can do that. Does it make sense to rename `routes` -> `httpRoutes` or it'll be a huge breaking change?

Reply:
> Sure, if you think that's cleaner then we can do that

Yep, I think it will be cleaner, and we will get type safety for free.

> or it'll be a huge breaking change?

Right. We can't break this, so we'll have to live with `routes` meaning HTTP routes.

Reply:
Done the above changes as per suggestion. @jessesuen Could you please take another look?

Reply:
@agrawroh - I took a closer look at the Istio VirtualService spec and I think we are oversimplifying by making tlsRoutes only match on numeric ports. It turns out port is an optional field for TLSMatchAttributes, and the tls match could be against something else (e.g. SNI hosts). For example the user may have a VirtualService like:

```yaml
  tls:
  - match:
    - port: 443
      sniHosts:
      - login.bookinfo.com
    route:
    - destination:
        host: login.prod.svc.cluster.local
  - match:
    - port: 443
      sniHosts:
      - reviews.bookinfo.com
    route:
    - destination:
        host: reviews.prod.svc.cluster.local
```

In the above example, port 443 is used in both routes. The user may want to split traffic on `reviews.bookinfo.com` but not `login.bookinfo.com`. I think the spec has to change to accommodate this. Instead of a port list, we should do something like:

```yaml
          virtualService:
            name: rollouts-demo-vsvc
            routes:
            - primary
            tlsRoutes:
            - port: 443
              sniHosts:
              - reviews.bookinfo.com
```

Both port and sniHosts would be optional but the user has to pick at least one. NOTE that there are also other fields like gateways, sourceLabels, sourceNamespace which can match. We don't necessarily need to support all of them off the bat, but we need to leave room to support them in the future.

Reply:
Yeah, that make sense. I can add support for SNI in this PR. Do you think we need to support more things as of now or is it okay to do just port & SNI as a start.

Reply:
I think port + SNI will be sufficient for this PR.

Reply:
@jessesuen I pushed the change if you wanna take another look at it.